execute/factory_test.go

```diff
@@ -0,0 +1,221 @@
+package execute
+
+import (
+	"fmt"
+	"math/big"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/smartcontractkit/libocr/offchainreporting2plus/ocr3types"
+
+	"github.com/smartcontractkit/chainlink-ccip/execute/exectypes"
+	typeconv "github.com/smartcontractkit/chainlink-ccip/internal/libs/typeconv"
+	dt "github.com/smartcontractkit/chainlink-ccip/internal/plugincommon/discovery/discoverytypes"
+	"github.com/smartcontractkit/chainlink-ccip/pkg/consts"
+	"github.com/smartcontractkit/chainlink-ccip/pkg/reader"
+	"github.com/smartcontractkit/chainlink-ccip/pkg/types/ccipocr3"
+)
+
+/*
+200 reports, 200 messages
+-------------------------
+Total size of observation: 1056507
+200 CommitReports: 60635
+200 Messages: 583635
+TokenData: 112035
+Costly Messages: 13945
+Nonces: 12035
+Contracts: 274962
+
+900 reports, 100 messages
+-------------------------
+Total size of observation: 937407
+900 CommitReports: 302235
+100 Messages: 291835
+TokenData: 56035
+Costly Messages: 7045
+Nonces: 6035
+Contracts: 274962
+
+100 reports, 100 messages
+-------------------------
+Total size of observation: 665507
+100 CommitReports: 30335
+100 Messages: 291835
+TokenData: 56035
+Costly Messages: 7045
+Nonces: 6035
+Contracts: 274962
+
+100 reports, 200 messages
+-------------------------
+Total size of observation: 1025407
+100 CommitReports: 30335
+200 Messages: 582835
+TokenData: 112035
+Costly Messages: 13945
+Nonces: 12035
+Contracts: 274962
+
+100 reports, 350 messages
+-------------------------
+Total size of observation: 1565257
+100 CommitReports: 30335
+350 Messages: 1019335
+TokenData: 196035
+Costly Messages: 24295
+Nonces: 21035
+Contracts: 274962
+*/
+func TestObservationSize(t *testing.T) {
+	t.Skip("This test is for estimating message sizes, not for running in CI")
```

Comment:
In the PR description you mention 350 messages fill the observation limit. 
1. Where do we get the observation limit from? Is it dictated by OCR protocl?
2. In all cases I think having a test to assert that 350 messages (or whatever the limit we set) doesn't exceed the observation limit to prevent issues in the future.

Reply:
350 messages is a worst case with 1000 bytes of data and 250 bytes of token data. This doesn't even account for the discovery observation or commit reports. For small messages with one token transfer and no token data we can fit over 1100 messages into a single observation.

All of this is totally informational to help us choose a reasonable number. We would need metrics about what the average message looks like in order to figure out the expected throughput, for example:
* token only and how many tokens.
* data only and how much data.
* both tokens and data.
* number of messages per commit report.
* frequency of messages requiring token data.